examples/gaussian_process/plot_gpr_prior_posterior.py

```diff
@@ -196,23 +196,30 @@ def plot_gpr_samples(gpr_model, n_samples, ax):
 # %%
 # Dot-product kernel
 # ..................
+from sklearn.gaussian_process import GaussianProcessRegressor
 from sklearn.gaussian_process.kernels import ConstantKernel, DotProduct
+from sklearn.preprocessing import StandardScaler
+import matplotlib.pyplot as plt
 
-kernel = ConstantKernel(0.1, (0.01, 10.0)) * (
-    DotProduct(sigma_0=1.0, sigma_0_bounds=(0.1, 10.0)) ** 2
-)
-gpr = GaussianProcessRegressor(kernel=kernel, random_state=0)
+# Scale the data
+scaler = StandardScaler()
+X_train_scaled = scaler.fit_transform(X_train)
 
-fig, axs = plt.subplots(nrows=2, sharex=True, sharey=True, figsize=(10, 8))
+# Define the kernel
+kernel = ConstantKernel(0.1, (0.01, 10.0)) * (DotProduct(sigma_0=1.0, sigma_0_bounds=(0.1, 10.0)) ** 2)
 
-# plot prior
+# Increase the number of iterations
+gpr = GaussianProcessRegressor(kernel=kernel, random_state=0, n_restarts_optimizer=10, optimizer='fmin_l_bfgs_b')
```

Comment:
Adding comments to the code is mostly required when showing the reasoning behind a complex decision, for instance here, a comment would rather explain the need for the "fmin_l_bfgs_b" optimizer.

Also, try to keep lines under 88 characters (see the linter comment on this PR).